### PR TITLE
Fix startup on Python 3.12

### PR DESCRIPTION
**Pull request by luk1337**
Check-list
----------

 * [ ] There is a linked issue discussing the motivations for this feature or bugfix
 * [ ] Unit tests have been added where possible
 * [ ] I've added / updated documentation for any user-facing features.
 * [ ] Performance seems to be comparable or better than current `main`

What this change is adding / fixing
-----------------------------------

This makes QuodLibet work with Python 3.12.

### CODE REVIEW

**Comment by luk1337** [comment]
> Plugins don't load.

`importlib.util._find_spec()` works but is hacky...

File: quodlibet/util/importhelper.py [hunk 1]
```diff
@@ -92,16 +92,16 @@ def load_module(name, package, path):
     except KeyError:
         pass
 
-    loader = importlib.find_loader(fullname, [path])
-    if loader is None:
+    spec = importlib.util._find_spec(fullname, [path])
```
Comment (luk1337) [inline quodlibet/util/importhelper.py:95]: Any idea how we can avoid calling private method here?

File: quodlibet/_import.py [hunk 2]
```diff
@@ -31,6 +32,19 @@ def __init__(self, name, packages):
         self._name = name
         self._packages = packages
 
+    def find_spec(self, fullname, path, target=None):
```
[the remark this answers is not in the document]
Comment (luk1337) [reply quodlibet/_import.py:35]: Now it's ok

File: quodlibet/util/importhelper.py (hunk 1 above)
Comment (luk1337) [reply quodlibet/util/importhelper.py]: This is also done.

**Comment by luk1337** [comment]
This is ready for review.

**Comment by lazka** [comment]
thanks

**Comment by luk1337** [comment]
Rebased since fbc3260282645403dc2bc7c248bb584b27f0958f pulled some of my changes.

**Comment by lazka** [comment]
This broke the loading of the titlecase plugin for some reason

**Comment by luk1337** [comment]
> This broke the loading of the titlecase plugin for some reason

This works for me with Python 3.12.2 and quodlibet#main on Linux.

**Comment by lazka** [comment]
It depends on the order the plugins are loaded, which is random.

Likely fixed in 7b247a5e0c39006c089efd38ca37e80b527baa23

**Pull request merged**